Linux build FileNotFoundError - docs/DXIL.rst.tmp

I noticed a new build error in DXC with the latest July 2022 1.7.2207 release that was not present in the April 2021 1.6.2104 release.
```
Traceback (most recent call last):
  File "<dxc-src-dir>/utils/hct/hctgen.py", line 183, in <module>
    exit(writeCodeTag(args))
  File "<dxc-src-dir>/utils/hct/hctgen.py", line 45, in writeCodeTag
    out = openOutput(args)
  File "<dxc-src-dir>/utils/hct/hctgen.py", line 55, in openOutput
    return open(args.output, 'w', newline=getNewline(args))
FileNotFoundError: [Errno 2] No such file or directory: '<dxc-build-dir>/docs/DXIL.rst.tmp'
make[2]: *** [docs/DXIL.rst.tmp] Error 1
make[1]: *** [CMakeFiles/DxilDocs.dir/all] Error 2
```
I can workaround it by doing `mkdir <dxc-build-dir>/docs` i.e. creating the output docs folder in the build dir.

--- a/utils/hct/hctgen.py
+++ b/utils/hct/hctgen.py
@@ -52,6 +52,9 @@ def writeCodeTag(args):
   return 0
   
 def openOutput(args):
+  outputDir = os.path.dirname(os.path.realpath(args.output))
+  if not os.path.exists(outputDir):
+    os.makedirs(outputDir)
   return open(args.output, 'w', newline=getNewline(args))
 
 def printHeader(out, filename):

